File: PostoGasolina.Business/Models/Veiculo.cs
using PostoGasolina.Business.Models.Enums;
using System;
using System.Collections.Generic;

namespace PostoGasolina.Business.Models
{
    public class Veiculo : Entity
    {
        public Guid ClienteId { get; set; }
        public string Marca { get; set; }
        public string Modelo { get; set; }
        public int Ano { get; set; }
        public string Placa { get; set; }
        public IdTipoCombustivel TipoCombustivelId { get; set; }
        public TipoCombustivel TipoCombustivel { get; set; }
        public bool Ativo { get; set; }
        public Cliente Cliente { get; set; }
        public IEnumerable<Abastecimento> Abastecimentos { get; set; }

    }
}

File: PostoGasolina.App/ExtJS/app/model/Veiculo.js
Ext.define('PostoGasolina.model.Veiculo', {
    extend: 'Ext.data.Model',

    fields: [
        { name: 'id' },
        { name: 'marca', type: 'string' },
        { name: 'modelo', type: 'string' },
        { name: 'ano', type: 'string' },
        { name: 'placa', type: 'string' },
        { name: 'ativo', type: 'bool' },
        { name: 'clienteid', mapping: 'cliente.id', type: 'guid' },
        { name: 'cliente_nome', mapping: 'cliente.nome', type: 'string' },
        { name: 'cliente_documento', mapping: 'cliente.documento', type: 'string' },
        { name: 'tipoCombustivelid', mapping: 'tipoCombustivel.id', type: 'int' }, 
        { name: 'tipoCombustivel_descricao', mapping: 'tipoCombustivel.descricao', type: 'string' },
        { name: 'tipoCombustivel', mapping: function (o) { return o.tipoCombustivel ? o.tipoCombustivel : null; } }
    ]

});